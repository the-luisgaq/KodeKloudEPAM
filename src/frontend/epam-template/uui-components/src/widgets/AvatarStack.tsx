import * as React from 'react';
import cx from 'classnames';
import type { IHasCX, IHasRawProps } from '@epam/uui-core';
import { Avatar } from './Avatar';
import css from './AvatarStack.module.scss';

export interface AvatarStackProps extends IHasCX, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    avatarSize: '24' | '36' | '48' | '144';
    urlArray: string[];
    direction: 'right' | 'left';
    avatarsCount?: number;
    renderItem?: (url: string) => React.ReactElement<any>;
}

export const AvatarStack = React.forwardRef<HTMLDivElement, AvatarStackProps>((props, ref) => {
    const {
        avatarSize, urlArray, direction, avatarsCount, renderItem,
    } = props;

    const firstElements = avatarsCount && urlArray.length > avatarsCount ? urlArray.slice(0, avatarsCount) : urlArray;
    const styleObj = { '--overlap': `-${+avatarSize / 4}px` } as object;
    return (
        <div className={ cx(props.cx, css.root) } ref={ ref } { ...props.rawProps }>
            <div role="group" style={ styleObj } className={ cx('uui-avatar_stack-avatars', css.container, css['avatar-' + direction]) }>
                {firstElements.map((avatar, index) => {
                    return renderItem ? (
                        <React.Fragment key={ index }>{renderItem(avatar)}</React.Fragment>
                    ) : (
                        <Avatar key={ index } size={ avatarSize } img={ avatar } alt="avatar" />
                    );
                })}
            </div>
            <div className="uui-avatar_stack-count">{avatarsCount && urlArray.length > avatarsCount ? '+' + (urlArray.length - avatarsCount) : null}</div>
        </div>
    );
});
