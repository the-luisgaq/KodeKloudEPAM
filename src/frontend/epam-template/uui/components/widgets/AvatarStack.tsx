import { AvatarStack as uuiAvatarStack, AvatarStackProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import css from './AvatarStack.module.scss';

export const AvatarStack = withMods<AvatarStackProps, AvatarStackProps>(uuiAvatarStack, () => [css.root]);
