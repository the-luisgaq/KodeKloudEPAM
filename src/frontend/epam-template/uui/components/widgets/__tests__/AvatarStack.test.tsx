import React from 'react';
import { render } from '@epam/uui-test-utils';
import { AvatarStack } from '../AvatarStack';

describe('AvatarStack', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(
            <AvatarStack
                avatarsCount={ 3 }
                urlArray={ Array(5).fill('https://api.dicebear.com/7.x/pixel-art/svg?seed=Coco&radius=50&backgroundColor=b6e3f4') }
                avatarSize="36"
                direction="right"
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
