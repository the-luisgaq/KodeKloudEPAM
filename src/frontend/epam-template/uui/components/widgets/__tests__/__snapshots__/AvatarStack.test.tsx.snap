// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarStack should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root root"
  >
    <div
      class="uui-avatar_stack-avatars container avatar-right"
      role="group"
      style="--overlap: -9px;"
    >
      <img
        alt="avatar"
        class="avatar"
        height="36"
        src="https://api.dicebear.com/7.x/pixel-art/svg?seed=Coco&radius=50&backgroundColor=b6e3f4"
        width="36"
      />
      <img
        alt="avatar"
        class="avatar"
        height="36"
        src="https://api.dicebear.com/7.x/pixel-art/svg?seed=Coco&radius=50&backgroundColor=b6e3f4"
        width="36"
      />
      <img
        alt="avatar"
        class="avatar"
        height="36"
        src="https://api.dicebear.com/7.x/pixel-art/svg?seed=Coco&radius=50&backgroundColor=b6e3f4"
        width="36"
      />
    </div>
    <div
      class="uui-avatar_stack-count"
    >
      +2
    </div>
  </div>
</DocumentFragment>
`;
